--- src/store.rs
+++ src/store.rs
@@ -176,8 +176,8 @@
         Ok(Self {
             file_path,
             inner: Arc::new(Mutex::new(FileInnerState {
-                tx,
                 next_global_seq,
+                tx,
                 id_index,
             })),
         })
@@ -289,7 +289,7 @@
                     }
                 }

-                let mut reader = state.unwrap();
+                let mut reader = state.expect("State should not be None");
                 loop {
                     let mut len_buf = [0u8; 4];
                     match reader.read_exact(&mut len_buf).await {
